website/docs/configuration/overview.html.md

```diff
@@ -0,0 +1,182 @@
+infrastructure object, while a module might describe a set of objects and the
+necessary relationships between them in order to create a higher-level system.
+
+A _Terraform configuration_ consists of a _root module_, where evaluation
+begins, along with a tree of child modules created when one module calls
+another.
+
+## Arguments, Blocks, and Expressions
+
+The syntax of the Terraform language consists of only a few basic elements:
+
+```hcl
+resource "aws_vpc" "main" {
+  cidr_block = var.base_cidr_block
+}
+
+<BLOCK TYPE> "<BLOCK LABEL>" "<BLOCK LABEL>" {
+  # Block body
+  <IDENTIFIER> = <EXPRESSION> # Argument
+}
+```
+
+- _Blocks_ are containers for other content and usually represent the
+  configuration of some kind of object, like a resource. Blocks have a
+  _block type,_ can have zero or more _labels,_ and have a _body_ that contains
+  any number of arguments and nested blocks. Most of Terraform's features are
+  controlled by top-level blocks in a configuration file.
+- _Arguments_ assign a value to a name. They appear within blocks.
+- _Expressions_ represent a value, either literally or by referencing and
+  combining other values. They appear as values for arguments, or within other
+  expressions.
+
+For full details about Terraform's syntax, see:
+
+- [Configuration Syntax](./syntax.html)
+- [Expressions](./expressions.html)
+
+## Code Organization
+
+The Terraform language uses configuration files that are named with the `.tf`
+file extension. There is also [a JSON-based variant of the language](./syntax-json.html)
+that is named with the `.tf.json` file extension.
+
+Configuration files must always use UTF-8 encoding, and by convention are
+usually maintained with Unix-style line endings (LF) rather than Windows-style
+line endings (CRLF), though both are accepted.
+
+A _module_ is a collection of `.tf` or `.tf.json` files kept together in a
+directory. The root module is built from the configuration files in the
+current working directory when Terraform is run, and this module may reference
+child modules in other directories, which can in turn reference other modules,
+etc.
+
+The simplest Terraform configuration is a single root module containing only
+a single `.tf` file. A configuration can grow gradually as more resources
+are added, either by creating new configuration files within the root module
+or by organizing sets of resources into child modules.
+
+## Configuration Ordering
+
+Because Terraform's configuration language is declarative, the ordering of
+blocks is generally not significant. (The order of `provisioner` blocks within a
+resource is the only major feature where block order matters.)
+
+Terraform automatically processes resources in the correct order based on
+relationships defined between them in configuration, and so you can organize
+resources into source files in whatever way makes sense for your infrastructure.
+
+## Terraform CLI vs. Providers
```

Comment:
This also feels a bit weird in the language intro. I can kind of see what you're getting at, but I think the goal is to help users figure out where to look for resource docs?

Reply:
Yeah, this bit can probably go.